src/pages/receipts/csvFileReceipt.js

```diff
@@ -119,55 +119,58 @@ const csvFileButtonSubmit = () => {
         showAnimation();
 
         try {
-            const results = await getBSIQueryData(dateFilter);
-            document.getElementById("csvDateInput").value = '';
+            const results = await getSpecimensByReceivedDate(dateFilter);
             const modifiedResults = modifyBSIQueryResults(results.data);
             generateBSIqueryCSVData(modifiedResults);
             hideAnimation();
         } catch (e) {
             hideAnimation();
-            showNotifications({ title: 'Error', body: `Error fetching BSI Query Data -- ${e.message}` }, true);
+            showNotifications({ title: 'Error', body: `Error fetching BSI Query Data -- ${e.message}` });
         }
     });
 }
 
-const getCurrentDate = () => {
-  const currentDate = new Date().toLocaleDateString('en-CA');
-  return currentDate;
-}
-
-const getBSIQueryData = async (filter) => {
+const getSpecimensByReceivedDate = async (dateFilter) => {
     try {
         const idToken = await getIdToken();
-        const response = await fetch(`${baseAPI}api=queryBsiData&type=${filter}`, {
+        const response = await fetch(`${baseAPI}api=getSpecimensByReceivedDate&receivedTimestamp=${dateFilter}`, {
             method: "GET",
             headers: {
                 Authorization: "Bearer " + idToken,
             },
         });
 
         if (response.status !== 200) {
-            throw new Error(`Error fetching BSI Query Data. ${response.status}`);
+          throw new Error(`Error fetching specimens by received date. ${response.status}`);
         }
 
-        return await response.json();    
+        return await response.json();
     } catch (e) {
         console.error(e);
-        throw new Error(`Error fetching BSI Query Data: ${e.message}`);
+        throw new Error(`Error fetching specimens by received date: ${e.message}`);
     }
-};
+}
 
+const getCurrentDate = () => {
+  const currentDate = new Date().toLocaleDateString('en-CA');
+  return currentDate;
```

Comment:
nit: I think this can be a single return and does not need a variable declaration.
`return new Date().toLocaleDateString('en-CA');`

Reply:
Agree - updated. Thanks.